src/wyzeapy/services/update_manager.py

```diff
@@ -67,9 +67,10 @@ async def update_next(self):
             # First we get the next updater off the queue
             updater = heappop(self.updaters)
             # if the updater has been removed, pop the next and clear it from the removed updaters
-            while self.check_if_removed(updater):
-                updater = heappop(self.updaters)
-                self.removed_updaters.remove(updater)
+            if self.removed_updaters:
+                while self.check_if_removed(updater):
+                    self.removed_updaters.remove(updater)
+                    updater = heappop(self.updaters)
```

Comment:
Not necessary as it was working before but this stops calling the `check_if_removed `function on every update cycle if there's nothing to check.

Reply:
Good addition 👍